### PR TITLE
Add GH Actions workflow for running tests

### PR DESCRIPTION
**Pull request by cmccandless**

### CODE REVIEW

**Comment by cmccandless** [comment]
It appears this repo is old enough that GH Actions is not enabled by default. @iHiD may need to enable it.

**Comment by cmccandless** [comment]
Here is an example of this workflow running all tests and passing: https://github.com/cmccandless/c-test-runner/pull/1/checks?check_run_id=781020318

**Comment by cmccandless** [comment]
@bergjohan can you please review?

**Comment by bergjohan** [comment]
Thank you! I've never worked with GH actions, but I think it looks good :) Are you ready to merge?

**Comment by cmccandless** [comment]
The workflow hasn't run yet for this PR. I believe GitHub Actions has to be enabled by a user with the correct permissions.

Pinging @iHiD

**Comment by cmccandless** [comment]
If it's already been enabled since my last commit, I'll push an empty commit (to be squashed later) to force a CI run.

**Comment by iHiD** [comment]
Actions are enabled for this repo (they are fall all repos by default on Exercism). So I'm not sure why it's not run? (cc @erikschierboom @saschamann)

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
```
Comment (SaschaMann) [inline .github/workflows/main.yml:1]: This line might be causing issues

Comment (SaschaMann) [inline .github/workflows/main.yml:1]: This line might be causing issues

Comment (cmccandless) [reply .github/workflows/main.yml:1]: I can remove it, but the same line causes no issues in [this](https://github.com/exercism/generic-test-runner/blob/master/.github/workflows/main.yml) workflow.

Comment (cmccandless) [reply .github/workflows/main.yml:1]: Also, shouldn't Actions be running, but failing due to a workflow config parse failure?

**Comment by iHiD** [comment]
I think the issue is that this is on your fork, and actions don't run on forks? 

So do the actions need to be on master to run in the first place?

**Comment by cmccandless** [comment]
@iHiD I think you might have solved it.

For more info: https://github.community/t/run-a-github-action-on-pull-request-for-pr-opened-from-a-forked-repo/16054

**Comment by cmccandless** [comment]
@bergjohan If you're willing to merge now, I can assist with debugging to make sure it is running smoothly after merge.

**Comment by bergjohan** [comment]
Thanks everyone, I'm merging this, let's see what happens :)

**Comment by bergjohan** [comment]
Looks like it worked! :) https://github.com/exercism/c-test-runner/runs/797233706

**Comment by iHiD** [comment]
Wahoo. Thanks all :)

**Comment by ErikSchierboom** [comment]
Awesome work!

**Pull request merged**